tests/rbd/readonly_namespace_image_migration.py

```diff
@@ -0,0 +1,149 @@
+"""Module to execute Live migration feature on pool namespace.
+
+Pre-requisites :
+We need cluster configured with atleast one client node with ceph-common package,
+conf and keyring files
+
+Test cases covered :
+CEPH-83573341 - Set read only on namespace and migrate the images and
+verify if migration is successful
+
+Test Case Flow :
+1. Create pool and an image
+2. Create a namespace within the pool
+3. Give readonly access to the namespace
+4. Create images within the namespace
+5. Migrate the created images to other pool
+6. verify for image migration
+"""
+
+from tests.rbd.exceptions import RbdBaseException
+from tests.rbd.rbd_utils import (
+    Rbd,
+    initial_rbd_config,
+    verify_migration_commit,
+    verify_migration_state,
+    verify_namespace_create,
+    verify_namespace_image_exit,
+)
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def image_migrate_and_verify(
+    rbd,
+    src_pool,
+    dest_pool,
+    namespace,
+    image,
+):
+    """Migrate the image from sorce to destination and verify migration
+
+    Args:
+        src_pool: source pool name
+        dest_pool: destination pool name
+        namespace: name of namespace
+    """
+    src_spec1 = src_pool + "/" + namespace + "/" + image
+    dest_spec1 = dest_pool + "/" + image
+    cmd1 = "ceph auth caps client.1"
+    cmd1 += " mon 'allow *' mgr 'allow profile rbd-read-only'"
+    cmd1 += f" osd 'profile rbd-read-only pool={src_pool}"
+    cmd1 += " namespace=testnamespace' -o /etc/ceph/ceph.client.1.keyring"
+    rbd.exec_cmd(cmd=cmd1)
```

Comment:
The lines are just limiting access of client 1 to the namespace
In further steps we are performing steps as client admin IIRC

"Set read only on namespace" and  "images with only read permissions" Needs t further examination and confirmation
on whether we need to implement it using only client acls or there is a way in rbd layer.

Reply:
Need to discuss more on this.

Reply:
@rkhambad Can you please update findings based on the the offline discussions last week ?

Reply:
Hi @VasishtaShastry , there is no such setting to make namespace as read-only, we have to use cephx-caps like we discuss and also explore command to run with particular client id.
We can use extra parameter as -n client.id along with the exec_cmd.